packages/gatsby/src/redux/actions/__tests__/internal.ts

```diff
@@ -0,0 +1,88 @@
+import { setSiteConfig } from "../internal"
+import reporter from "gatsby-cli/lib/reporter"
+
+jest.mock(`gatsby-cli/lib/reporter`, () => {
+  return {
+    panic: jest.fn(),
+  }
+})
+
+beforeEach(() => {
+  reporter.panic.mockClear()
+})
+
+describe(`setSiteConfig`, () => {
+  it(`let's you add a config`, () => {
+    const action = setSiteConfig({
+      siteMetadata: {
+        hi: true,
+      },
+    })
+    expect(action).toMatchInlineSnapshot(`
+      Object {
+        "payload": Object {
+          "pathPrefix": "",
+          "polyfill": true,
+          "siteMetadata": Object {
+            "hi": true,
+          },
+        },
+        "type": "SET_SITE_CONFIG",
+      }
+    `)
+  })
+
+  it(`handles empty configs`, () => {
+    const action = setSiteConfig()
+    expect(action).toMatchInlineSnapshot(`
+      Object {
+        "payload": Object {
+          "pathPrefix": "",
+          "polyfill": true,
+        },
+        "type": "SET_SITE_CONFIG",
+      }
+    `)
+  })
+
+  it(`Validates configs with unsupported options`, () => {
+    setSiteConfig({
+      someRandomThing: `hi people`,
+      plugins: [],
+    })
+
+    expect(reporter.panic).toBeCalledWith({
+      id: "10122",
+      context: {
+        sourceMessage: '"someRandomThing" is not allowed',
+      },
+    })
+  })
+
+  it(`It corrects pathPrefixes without a forward slash at beginning`, () => {
+    const action = setSiteConfig({
+      pathPrefix: `prefix`,
+    })
+
+    expect(action.payload.pathPrefix).toBe(`/prefix`)
+  })
+
+  it(`It removes trailing forward slash`, () => {
+    const action = setSiteConfig({
+      pathPrefix: `/prefix/`,
+    })
+    expect(action.payload.pathPrefix).toBe(`/prefix`)
+  })
+
+  it(`It removes pathPrefixes that are a single forward slash`, () => {
+    const action = setSiteConfig({
+      pathPrefix: `/`,
+    })
+    expect(action.payload.pathPrefix).toBe(``)
+  })
+
+  it(`It sets the pathPrefix to an empty string if it's not set`, () => {
+    const action = setSiteConfig({})
+    expect(action.payload.pathPrefix).toBe(``)
+  })
+})
```

Comment:
Those are tests that previously were reducer tests adjusted to be action creator test (same scenarios)